src/Objects.cpp

```diff
@@ -126,7 +127,7 @@ bool Triangle::intersect(const Ray& ray, Intersection& result) const {
     if (contains(pointIntersectingPlane)) {
         result.t      = t;
         result.point  = pointIntersectingPlane;
-        result.normal = planeNormal_;
+        result.normal = (Math::dot(ray.origin, planeNormal_) > 0.0f ? 1.0f : -1.0f) * planeNormal_;
```

Comment:
Might make sense to move this somewhere else since it's common for all intersections